src/KeyringController.ts

```diff
@@ -273,27 +327,28 @@ class KeyringController extends EventEmitter {
    *
    * Only supports 'Simple Key Pair'.
    *
-   * @param {string} type - The key pair type to check for.
-   * @param {Array<string>} newAccountArray - Array of new accounts.
-   * @returns {Promise<Array<string>>} The account, if no duplicate is found.
+   * @param type - The key pair type to check for.
+   * @param newAccountArray - Array of new accounts.
+   * @returns The account, if no duplicate is found.
    */
-  async checkForDuplicate(type, newAccountArray) {
+  async checkForDuplicate(
+    type: string,
+    newAccountArray: Hex[],
+  ): Promise<string[]> {
     const accounts = await this.getAccounts();
 
     switch (type) {
-      case KEYRINGS_TYPE_MAP.SIMPLE_KEYRING: {
+      case KeyringType.Simple: {
         const isIncluded = Boolean(
           accounts.find(
             (key) =>
               key === newAccountArray[0] ||
-              key === stripHexPrefix(newAccountArray[0]),
+              key === stripHexPrefix(newAccountArray[0] as Hex),
```

Comment:
How do we know that `newAccountArray[0]` satisfies `Hex`? Instead of typecasting this, is there a function we can define/use that takes a string and returns a Hex?

Reply:
[e36c134](https://github.com/MetaMask/KeyringController/pull/202/commits/e36c1348aadc882625deda681912d6b45786060e)